.github/workflows/soundness.yml

```diff
@@ -63,6 +63,10 @@ on:
         type: boolean
         description: "Boolean to enable the Python lint check job. Defaults to true."
         default: true
+      linux_pre_run_command:
```

Comment:
Should this really be executed before every soundness check. It seems like this is only relevant for the ones that need to build something e.g. docs and api break

Reply:
I wondered that myself but I think it's plausible that a caller may want to do some other source preparations, munging, generation that this would permit.

Reply:
I'm not set on it being this way though if you'd rather we solve the immediate requirement.

Reply:
swift-java needs a bunch of weird stuff, like installing a JDK for builds to work.

I agree that it is only for things which "build" the formatting check for example does no need it. At least so far AFAICS

Reply:
Yep, I'm happy taking this approach. Thanks for the feedback everyone.